src/adiv5.c

```diff
@@ -32,16 +32,181 @@
 #define DO_RESET_SEQ 0
 #endif
 
+/* All this should probably be defined in a dedicated ADIV5 header, so that they
+ * are consistently named and accessible when needed in the codebase.
+ */
+
 /* ROM table CIDR values */
-#define CIDR_ROM_TABLE  0xb105100d
-#define CIDR_GENERIC_IP 0xb105e00d
-#define CIDR_DEBUG      0xb105900d
+#define CIDR0_OFFSET    0xFF0 /* DBGCID0 */
+#define CIDR1_OFFSET    0xFF4 /* DBGCID1 */
+#define CIDR2_OFFSET    0xFF8 /* DBGCID2 */
+#define CIDR3_OFFSET    0xFFC /* DBGCID3 */
+
+/* Component class ID register can be broken down into the following logical
+ * interpretation of the 32bit value consisting of the least significant bytes
+ * of the 4 CID registers:
+ * |7   ID3 reg   0|7   ID2 reg   0|7   ID1 reg   0|7   ID0 reg   0|
+ * |1|0|1|1|0|0|0|1|0|0|0|0|0|1|0|1| | | | |0|0|0|0|0|0|0|0|1|1|0|1|
+ * |31           24|23           16|15   12|11     |              0|
+ * \_______________ ______________/\___ __/\___________ ___________/
+ *                 V                   V               V
+ *             Preamble            Component       Preamble
+ *                                   Class
+ * \_______________________________ _______________________________/
+ *                                 V
+ *                           Component ID
+ */
+#define CID_PREAMBLE    0xB105000D
+#define CID_CLASS_MASK  0x0000F000
+#define CID_CLASS_SHIFT 12
+/* The following enum is based on the Component Class value table 13-3 of the
+ * ADIv5 standard.
+ */
+enum cid_class {
+	cidc_gvc = 0x0,    /* Generic verification component*/
+	cidc_romtab = 0x1, /* ROM Table, std. layout (ADIv5 Chapter 14) */
+	/* 0x2 - 0x8 */    /* Reserved */
+	cidc_dc = 0x9,     /* Debug component, std. layout (CoreSight Arch. Spec.) */
+	/* 0xA */          /* Reserved */
+	cidc_ptb = 0xB,    /* Peripheral Test Block (PTB) */
+	/* 0xC */          /* Reserved */
+	cidc_dess = 0xD,   /* OptimoDE Data Engine SubSystem (DESS) component */
+	cidc_gipc = 0xE,   /* Generic IP Component */
+	cidc_pcp = 0xF,    /* PrimeCell peripheral */
+	cidc_unknown = 0x10
+};
 
-#define PIDR_REV_MASK 0x0FFF00000ULL
-#define PIDR_ARMv7M   0x4000BB000ULL
-#define PIDR_ARMv6M   0x4000BB008ULL
-#define PIDR_ARMv7MF  0x4000BB00CULL
-#define PIDR_ARMv7A   0x4000BBC09ULL
+#ifdef PLATFORM_HAS_DEBUG
+/* The reserved ones only have an R in them, to save a bit of space. */
+static const char const *cidc_debug_strings[] =
+{
+	"Generic verification component",           /* 0x0 */
```

Comment:
the remaining strings should be in there as they are usually reserved, I wanted to prevent the debug to output just an empty string when the component class is one of them.